### PR TITLE
changes for validating org external id with channel

### PR DESCRIPTION
**Pull request by rahul-tarento**

### CODE REVIEW

File: actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java [hunk 1]
```diff
@@ -254,6 +254,21 @@ private void createUser(Request actorMessage) {
         return;
       }
     }
+    if (userMap.containsKey(JsonKey.ORG_EXTERNAL_ID)) {
```
Comment (amolnin) [inline actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java]: Check and throw error if org_external_id and organisationid both are submitted

File: actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java [hunk 2]
```diff
@@ -254,6 +254,21 @@ private void createUser(Request actorMessage) {
...
+      String channel = (String) userMap.get(JsonKey.CHANNEL);
+      Organisation org = organisationClient.esGetOrgByExternalId(orgExternalId, channel);
+      if (org == null) {
+        ProjectLogger.log(
+            "UserManagementActor:createUser: No organisation with orgExternalId = "
+                + orgExternalId
+                + " and channel = "
+                + channel,
+            LoggerEnum.ERROR.name());
+        ProjectCommonException.throwClientErrorException(ResponseCode.invalidOrgData);
```
Comment (bvinayakumar) [inline actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java]: Throw invalidParameterValue.

File: actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java (hunk 1 above)
Comment (bvinayakumar) [reply actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java]: You can use parameterMismatch error code.

File: actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java [hunk 3]
```diff
@@ -254,6 +256,41 @@ private void createUser(Request actorMessage) {
...
+      String orgId =
+          orgExternalService.getOrgIdFromOrgExternalIdAndProvider(orgExternalId, channel);
+      if (StringUtils.isBlank(orgId)) {
+        ProjectLogger.log(
+            "UserManagementActor:createUser: No organisation with orgExternalId = "
+                + orgExternalId
+                + " and channel = "
+                + channel,
+            LoggerEnum.ERROR.name());
+        throw new ProjectCommonException(
```
Comment (amolnin) [inline actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java]: use static funciton of projectcommonexception

**Pull request merged**